docs/async_service.rst

```diff
@@ -172,6 +172,11 @@ If a task raises an exception it will trigger cancellation of the service.
 Upon exiting, all errors that were encountered while running the service will
 be re-raised.
 
+For slighly nicer logging output you can provide a ``name`` as a keyword
+argument to `~async_service.abc._InternalManagerAPI.run_task` which will be
```

Comment:
Hm, it looks like this is a private API, but we're documenting it publicly. Should it become a public API?

Reply:
Agreed. I think `_InternalManagerAPI` should be renamed and drop the underscore. As I understand it, we want to make sure that external consumers are only exposed to `ManagerAPI` which is fine. We can ensure to use `ManagerAPI` as the return type to hide the fact that we are returning an `_InternalManagerAPI`. So, if someone would want to call APIs on it they would be forced to `cast()` it down to the underlying `_InternalManagerAPI`.

But `_InternalManagerAPI` is still used from *within* the service so to the *internals* of the service there's nothing private about `_InternalManagerAPI`. I guess that's true for any class, there's always someone consuming a class which means it can not be private on the class level.

I think `_InternalManagerAPI` should just become `InternalManagerAPI`

Reply:
correct and :+1: